service/storage_service.cc

```diff
@@ -5007,6 +5007,16 @@ std::map<token, inet_address> storage_service::get_token_to_endpoint_map() {
     return result;
 }
 
+std::map<token, inet_address> storage_service::get_tablet_to_endpoint_map(table_id table) {
+    const auto& tm = get_token_metadata();
+    const auto& tmap = tm.tablets().get_tablet_map(table);
+    std::map<token, inet_address> result;
+    for (std::optional<locator::tablet_id> tid = tmap.first_tablet(); tid; tid = tmap.next_tablet(*tid)) {
+        result.emplace(tmap.get_last_token(*tid), tm.get_endpoint_for_host_id(tmap.get_primary_replica(*tid)));
+    }
```

Comment:
potential source of stall? if each iteration takes 0.1us, and we have 100k tablets, that's 10ms. even 10k tablets would bring us beyond the task quota.

Reply:
Added yield